flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java

```diff
@@ -375,6 +375,8 @@ public void invoke(IN value, SinkFunction.Context context) throws Exception {
 
 	@Override
 	public void close() throws Exception {
-		buckets.close();
+		if (buckets != null) {
```

Comment:
`bucket` is non-final, and `Cmd + Alt + F7` shows there is only one assignment in `initializeState()`. One reason why `buckets` can be `null` is that `initializeState()` is not called at all. Another reason is that `initializeState()` exits prematurely due to exceptions. 
If there were more things to be closed, we would need more null checks, and I would not add a comment to every null check. Hence, I would also not add a comment when there is only a single null check. Imo a comment could be perceived as too verbose because the necessity of it is simple enough to understand.